### PR TITLE
refactor(flexlayout): update flex layout css to reflect toolbar spacing

### LINKED ISSUE #1690: Color contrast issues in flex layout examples
We should address these so they don't fail a11y.

```
1 error found in: /layouts/Flex/examples-full/

1) Elements must have sufficient color contrast -- https://dequeuniversity.com/rules/axe/3.1/color-contrast -- impact: serious
      
-------html violation instances-------

<h2 class="example-title">
  Basic flex - <code>.pf-l-flex</code>
</h2>
<code>.pf-l-flex</code>
<h2 class="example-title">
  Flex with modified spacing - <code>.pf-l-flex.pf-m-space-items-xl</code>
</h2>
<code>.pf-l-flex.pf-m-space-items-xl</code>
<h2 class="example-title">
  Flex with modified spacing - <code>.pf-l-flex.pf-m-space-items-none</code>
</h2>
<code>.pf-l-flex.pf-m-space-items-none</code>
<h2 class="example-title">
  Individually spaced items.</h2>
<h2 class="example-title">
  Test 1</h2>
<h2 class="example-title">
  Test 2</h2>
<h2 class="example-title">
  Nested flex items</h2>
<h2 class="example-title">
  Mixed nesting flex items</h2>
<h2 class="example-title">
  Mixed nesting flex items</h2>
<h2 class="example-title">
  Inception grouping</h2>
<h2 class="example-title">
  Aligning right with align right. This solution will always align element right by setting margin-left: auto, including when wrapped.</h2>
<h2 class="example-title">
  Align right on single groups.</h2>
<h2 class="example-title">
  Align right on multiple groups.</h2>
<h2 class="example-title">
  Using <code>.pf-m-flex-grow</code> + <code>.pf-m-nowrap</code>.</h2>
<code>.pf-m-flex-grow</code>
<code>.pf-m-nowrap</code>
<h2 class="example-title">
  Using <code>.pf-m-flex-grow</code> + <code>.pf-m-nowrap</code> (responsive).</h2>
<code>.pf-m-flex-grow</code>
<code>.pf-m-nowrap</code>
<h2 class="example-title">
  justify-content flex end.</h2>
<h2 class="example-title">
  justify-content space-between.</h2>
<h2 class="example-title">
  justify-contenting flex start (use to reset flex-direction to default).</h2>
<h2 class="example-title">
  Column flex - <code>.pf-l-flex.pf-m-column</code>
</h2>
<code>.pf-l-flex.pf-m-column</code> 
```

### PR DESCRIPTION
**Pull request by mattnolting**
refers to #1528 
closes #1690 

This pr includes refactored flex layout documentation and updated css support. The updates address the request for default toolbar spacing in #1528. I'll apply the updated flex layout to the toolbar and open a separate PR, which will directly address #1528.

### CODE REVIEW

File: src/patternfly/layouts/Flex/flex.scss [hunk 1]
```diff
@@ -119,36 +118,25 @@ $pf-flex-spacer-map: (
...
-  &[class*="pf-m-justify-space-between"] {
-    > * {
+    &:last-child {
       --pf-l-flex--spacer: 0;
     }
   }
 
   // Reset item spacing in nested flex layouts
   & & {
+    --pf-l-flex--spacer: var(--pf-l-flex--flex--spacer);
```
Comment (mcoker) [inline src/patternfly/layouts/Flex/flex.scss]: Changing this default would be a breaking change. It's safer if you were to reduce the spacer, but increasing the spacer means lines may now break when they wouldn't have broken before. Can we retain the previous value?

File: src/patternfly/layouts/Flex/examples/flex-layout-example.hbs [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+    {{#> example-text example-text--modifier="example-border"}}Flex item{{/example-text}}
+    {{#> example-text example-text--modifier="example-border"}}Flex item{{/example-text}}
+  {{/l-flex}}
+  {{#> l-flex l-flex--modifier="example-border"}}
+    {{#> example-text example-text--modifier="example-border"}}Flex item{{/example-text}}
+  {{/l-flex}}
+{{/l-flex}}
+
+<br>
+{{#> example-title}}Adjusting width with <code>.pf-m-flex-1</code>. In this example, all <code>.pf-l-flex__group</code>s are set to <code>.pf-m-flex-1</code>. They will share available space equally.{{/example-title}}
```
Comment (mcoker) [inline src/patternfly/layouts/Flex/examples/flex-layout-example.hbs]: leftover reference to `.pf-l-flex__group`

File: src/patternfly/layouts/Flex/flex.scss [hunk 3]
```diff
@@ -167,41 +155,68 @@ $pf-flex-spacer-map: (
...
+      flex-wrap: wrap-reverse;
+    }
+
     &.pf-m-nowrap#{$mod} {
       flex-wrap: nowrap;
     }
 
-    &.pf-m-wrap-reverse#{$mod} {
-      flex-wrap: wrap-reverse;
+    &.pf-m-flex-none#{$mod} {
```
Comment (mcoker) [inline src/patternfly/layouts/Flex/flex.scss]: Can you add this to the documentation? You might also consider moving it by the other shorthand `flex` properties, like after `.pf-m-flex-default`

File: src/patternfly/layouts/Flex/docs/code.md [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+| `.pf-m-align-items-flex-end{-on-[breakpoint]}` | `.pf-l-flex` | Modifies align-items property to flex-end. |
+| `.pf-m-align-items-center{-on-[breakpoint]}` | `.pf-l-flex` | Modifies align-items property to center. |
+| `.pf-m-align-items-stretch{-on-[breakpoint]}` | `.pf-l-flex` | Modifies align-items property to stretch. |
+| `.pf-m-align-items-baseline{-on-[breakpoint]}` | `.pf-l-flex` | Modifies align-items property to baseline. |
+| `.pf-m-align-content-flex-start{-on-[breakpoint]}` | `.pf-l-flex` | Modifies align-content property to flex-start. |
+| `.pf-m-align-content-flex-end{-on-[breakpoint]}` | `.pf-l-flex` | Modifies align-content property to flex-end. |
+| `.pf-m-align-content-center{-on-[breakpoint]}` | `.pf-l-flex` | Modifies align-content property to center. |
+| `.pf-m-align-content-stretch{-on-[breakpoint]}` | `.pf-l-flex` | Modifies align-content property to stretch. |
+| `.pf-m-align-content-space-between{-on-[breakpoint]}` | `.pf-l-flex` | Modifies align-content property to space-between. |
+| `.pf-m-align-content-center{-on-[breakpoint]}` | `.pf-l-flex` | Modifies align-content property to center. |
```
Comment (mcoker) [inline src/patternfly/layouts/Flex/docs/code.md]: this is in here twice. you can remove this one.

File: src/patternfly/layouts/Flex/flex.scss [hunk 5]
```diff
@@ -296,9 +317,24 @@ $pf-flex-spacer-map: (
     }
 
     .pf-m-full-width#{$mod} {
-      --pf-l-flex--spacer: 0;
+      width: 100%;
```
Comment (mcoker) [inline src/patternfly/layouts/Flex/flex.scss]: can you add this one to the docs?

**Comment by mcoker** [comment]
In this example, I would expect there to be 16px between flex items and 24px between nested flex layouts

Screen Shot 2019-04-18 at 4 31 24 PM

Looks like specificity problems with `.pf-m-space-items-*`, unless this is expected?

Screen Shot 2019-04-18 at 4 39 36 PM

Can you add `pf-m-justify-content-space-around` to the docs?

File: src/patternfly/layouts/Flex/examples/flex-basic-example.hbs [hunk 6]
```diff
@@ -1,32 +1,40 @@
 <h2 class="example-title">
-  Basic flex - <code>.pf-l-flex</code>
+  Basic flex.
```
Comment (christiemolloy) [inline src/patternfly/layouts/Flex/examples/flex-basic-example.hbs]: can you rewrite as: Basic Flex - `.pf-l-flex`

**Pull request merged**